File: 02_ProgrammingFundamentals/_11_RegularExpressions/_02_Exercise/_06_ExtractEmails.java
package _02_ProgrammingFundamentals._11_RegularExpressions._02_Exercise;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class _06_ExtractEmails {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String input = scan.nextLine();

        // Add variables
        String regexEmail = "((?<=\\s)[a-zA-Z0-9]+([-.]\\w*)*@[a-zA-Z]+?([.-][a-zA-Z]*)*(\\.[a-z]{2,}))";
        Pattern patternEmail = Pattern.compile(regexEmail);
        Matcher matcherEmail = patternEmail.matcher(input);

        // Extract emails and print result
        while (matcherEmail.find()) {
            String email = matcherEmail.group(1);
            System.out.println(email);
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_09_RegularExpressions/_02_Exercise/_06_ExtractEmails/_06_ExtractEmails.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

public class _06_ExtractEmails
{
    public static void Main(string[] args)
    {
        // Read input value
        string input = Console.ReadLine();

        // Add variables
        string regexEmail = "((?<=\\s)[a-zA-Z0-9]+([-.]\\w*)*@[a-zA-Z]+?([.-][a-zA-Z]*)*(\\.[a-z]{2,}))";
        Regex patternEmail = new Regex(regexEmail);
        Match matcherEmail = patternEmail.Match(input);

        // Extract emails and print result
        while (matcherEmail.Success)
        {
            Console.WriteLine(matcherEmail.Value);
            matcherEmail = matcherEmail.NextMatch();
        }
    }
}
